src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -71,6 +73,7 @@ internal abstract partial class MemberSemanticModel : CSharpSemanticModel
             _speculatedPosition = speculatedPosition;
 
             _operationFactory = new Lazy<CSharpOperationFactory>(() => new CSharpOperationFactory(this));
+            _analyzedVariableTypesOpt = new ConcurrentDictionary<LocalSymbol, LocalSymbol>();
```

Comment:
>new ConcurrentDictionary [](start = 40, length = 24)

We should only allocate if `Compilation.NullableSemanticAnalysisEnabled`, as in iteration 3.